spec/delete_host_spec.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+require 'rails_helper' # spec_helper?
+require 'sidekiq/testing'
+
+describe DeleteHost do
+  let(:user) { FactoryBot.create(:user) }
+  let(:host) { FactoryBot.create(:host, org_id: user.account.org_id) }
+  let(:profile) { FactoryBot.create(:profile, :with_rules, account: user.account) }
+  let(:message) { { 'id' => host.id, 'type' => 'delete' } }
+  let!(:test_result) { FactoryBot.create(:test_result, profile: profile, host: host) }
+  let!(:rule_result) do
+    FactoryBot.create(:rule_result, rule: profile.rules.sample, test_result: test_result, host: host)
+  end
+
+  before do
+    logger = double(Rails.logger)
+    allow(logger).to receive(:info)
+    allow(logger).to receive(:debug)
+    allow(Sidekiq).to receive(:logger).and_return(logger)
+
+    DeleteHost.clear
+  end
+
+  it 'performs and logs deletion' do
+    expect(profile.hosts.count).not_to eql(0)
+    expect(profile.score).not_to eql(0.0)
+    DeleteHost.perform_async(message)
+
+    expect(profile.test_results.count).to eql(1)
+    expect(profile.rule_results.count).to eql(1)
+    expect(Rails.logger).to receive(:audit_success).with("Deleted related records for host #{host.id}")
+    DeleteHost.drain
+    expect(profile.test_results.count).to eql(0)
+    expect(profile.rule_results.count).to eql(0)
+
+    expect(profile.hosts.count).to eql(0)
+    expect(profile.reload.score).to eql(0.0)
+  end
+
+  context 'when multiple hosts exist under a profile' do
+    let(:message) { { 'id' => profile.hosts.first.id, 'type' => 'delete' } }
+
+    it 'deletes host and updates score and counters' do
+      test_result.destroy
+      rule_result.destroy
+
+      FactoryBot.create_list(:host, 3, org_id: user.account.org_id) do |host|
+        test_result = FactoryBot.create(:test_result, profile: profile, host: host)
+        FactoryBot.create(:rule_result, rule: profile.rules.sample, test_result: test_result, host: host)
+      end
+
+      expect(profile.score).not_to eql(0.0)
+      old_score = profile.score
+      expect(profile.policy.test_result_host_count).to eql(3)
+      DeleteHost.perform_async(message)
+
+      DeleteHost.drain
+      expect(profile.reload.score).not_to eql(old_score)
+      expect(profile.policy.test_result_host_count).to eql(2)
+    end
+  end
+
+  context 'when host to be deleted does not exist' do
+    let(:message) { { 'id' => 'invalid-id', 'type' => 'delete' } }
+
+    it 'fails silently' do
+      DeleteHost.perform_async(message)
+      expect(profile.hosts.count).to eql(1)
+      expect(profile.score).not_to eql(0.0)
+
+      DeleteHost.drain
+      expect(profile.hosts.count).to eql(1)
+      expect(profile.score).not_to eql(0.0)
+    end
+  end
+
+  context 'when deletion is not possible' do
+    let(:err_message) { "Failed to delete related records for host #{host.id}: StandardError" }
```

Comment:
Let blocks are not regular variables, please move this into the expectation

Reply:
I had to extract the log message into a let due to RuboCop complaining about the length of the line. (over 120 chars)